Dockerfile

```diff
@@ -13,72 +13,110 @@
+# string for the optional file default value, then this command would copy
+# the entire build context, which is not what we want.
+COPY download.sh ${HADOOP_FILE}* /tmp/
+
+RUN set -eux; \
+  download.sh "${HADOOP_FILE}" "hadoop.tar.gz" "hadoop/core/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz"; \
+  tar xzf hadoop.tar.gz -C /tmp/; \
+  mv /tmp/hadoop-$HADOOP_VERSION /opt/hadoop; \
+  rm -rf /opt/hadoop/share/doc/hadoop
+
+##
+## Zookeeper image. Download/copy and extract the Zookeeper installation.
+##
+FROM buildbase as zookeeper
+
+ARG ZOOKEEPER_VERSION=3.8.0 \
+  ZOOKEEPER_FILE=_NOT_SET
+# Copy a known file along with the optional files (that might not exist).
+# The known file, along with '*' for the optional file allows the command
+# to succeed even if the optional file does not exist. If we used an empty
+# string for the optional file default value, then this command would copy
+# the entire build context, which is not what we want.
+COPY download.sh ${ZOOKEEPER_FILE}* /tmp/
+
+RUN set -eux; \
+  download.sh "${ZOOKEEPER_FILE}" "zookeeper.tar.gz" "zookeeper/zookeeper-$ZOOKEEPER_VERSION/apache-zookeeper-$ZOOKEEPER_VERSION-bin.tar.gz"; \
+  tar xzf zookeeper.tar.gz -C /tmp/; \
+  mv /tmp/apache-zookeeper-$ZOOKEEPER_VERSION-bin /opt/zookeeper
+
+##
+## Accumulo image. Download/copy and extract the Accumulo installation, build native libs, and copy in properties.
+##
+FROM buildbase as accumulo
+
+ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk
+
+ARG ACCUMULO_VERSION=2.1.0 \
+  ACCUMULO_FILE=_NOT_SET
+# Copy a known file along with the optional files (that might not exist).
+# The known file, along with '*' for the optional file allows the command
+# to succeed even if the optional file does not exist. If we used an empty
+# string for the optional file default value, then this command would copy
+# the entire build context, which is not what we want.
+COPY download.sh ${ACCUMULO_FILE}* /tmp/
+
+RUN set -eux; \
+  download.sh "${ACCUMULO_FILE}" "accumulo.tar.gz" "accumulo/$ACCUMULO_VERSION/accumulo-$ACCUMULO_VERSION-bin.tar.gz"; \
+  tar xzf accumulo.tar.gz -C /tmp/; \
+  mv /tmp/accumulo-$ACCUMULO_VERSION*/ /opt/accumulo; \
   /opt/accumulo/bin/accumulo-util build-native
 
 ADD properties/ /opt/accumulo/conf/
 
-ENV HADOOP_HOME=/opt/hadoop ZOOKEEPER_HOME=/opt/zookeeper ACCUMULO_HOME=/opt/accumulo PATH="$PATH:/opt/accumulo/bin"
+##
+## Final image. Copy extracted/built installations for hadoop, zookeeper, and accumulo.
+## Also set environment variables and entrypoint.
+##
+FROM base
+
+ARG HADOOP_USER_NAME=accumulo
+ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk \
+  HADOOP_HOME=/opt/hadoop \
+  HADOOP_USER_NAME=$HADOOP_USER_NAME \
+  ZOOKEEPER_HOME=/opt/zookeeper \
+  ACCUMULO_HOME=/opt/accumulo \
+  PATH="$PATH:/opt/accumulo/bin"
+
+COPY --from=hadoop /opt/hadoop /opt/hadoop
+COPY --from=zookeeper /opt/zookeeper /opt/zookeeper
+COPY --from=accumulo /opt/accumulo /opt/accumulo
```

Comment:
I was going for two things in the other PR.  Make build faster and make the accumulo only layer smaller (so that push/pull to container reg is faster).   Will this result in a larger layer when just rebuilding an Accumulo snapshot?  Larger than the 30 to 40mb I was seeing in #22

Reply:
Not sure what copy will if the from image did not change.  If copy reuses cache layer when hadoop and zookeeper images don't change, then it seems like it will be good.

Reply:
Each COPY command creates a new layer, so it should be good. I confirmed by running [dive](https://github.com/wagoodman/dive) on it, and it shows 
<img width="812" alt="image" src="https://user-images.githubusercontent.com/5400335/191116248-d63eabb2-46bf-486c-ab1c-d21a044ec2ea.png">

Reply:
That hadoop layer is hefty.

Reply:
Also, I did try a couple combinations and COPY does seem to use the cached layer when you'd expect. e.g., if I change the accumulo version, then it used the cached layers for the base image, hadoop, and zookeeper. Same if I changed only Zookeeper (didn't try Hadoop since it takes so long to download).